omniledger/service/contracts.go

```diff
@@ -138,6 +171,63 @@ func (s *Service) ContractConfig(cdb CollectionView, tx Instruction, coins []Coi
 // ContractDarc accepts the following instructions:
 //   - Spawn - creates a new darc
 //   - Invoke.Evolve - evolves an existing darc
-func (s *Service) ContractDarc(cdb CollectionView, tx Instruction, coins []Coin) (sc []StateChange, c []Coin, err error) {
-	return nil, nil, errors.New("Not yet implemented")
+func (s *Service) ContractDarc(coll CollectionView, tx Instruction,
+	coins []Coin) ([]StateChange, []Coin, error) {
+	if tx.Invoke == nil {
```

Comment:
Every contract is starting with different formulations of this. Repeat after me: repetition is not good.

Maybe we could make:
```
func (i *Instruction) Require(itype InstructionType) error {
  switch itype {
    case Spawn:
      if tx.Invoke != nil || tx.Delete != nil {
        return errors.New("transaction must be a spawn")
     ...etc...
```